File: backend/app/schemas/habit_completion.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class HabitCompletionCreate(BaseModel):
    habit_id: int
    completion_date: date
    completed: bool


class HabitCompletionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    habit_id: int
    completion_date: date
    completed: bool
    completed_at: datetime | None

File: frontend/src/types/habitCompletion.ts
export type HabitCompletion = {
  id: number;
  habit_id: number;
  completion_date: string;
  completed: boolean;
  completed_at: string | null;
};

export type HabitCompletionCreate = {
  habit_id: number;
  completion_date: string;
  completed: boolean;
};